### ISSUE TITLE
Style/FormatStringToken false positive with formatted input and template style enforced

### ISSUE THREAD

**Issue by inkstak** [rubocop/rubocop#7900]
## Expected behavior

When enforcing the template style: 
```
Style/FormatStringToken:
  EnforcedStyle: template
```

and with the following code:
```
format('%<number>0.02f', value) }
```

Formatting input doesn't work with template style - [source](https://ruby-doc.org/core-2.7.0/Kernel.html#method-i-format)

> %<name>s style uses format style, but %{name} style doesn’t.

The cop should not report offenses or accept an new configuration option for formatted input.

I guess this is a similar issue reported in [this comment](https://github.com/rubocop-hq/rubocop/issues/5630#issuecomment-546450054)
 

## Actual behavior

It returns the offense :

```
Style/FormatStringToken: Prefer template tokens (like %{foo}) over annotated tokens (like %<foo>s).
  format('%<number>0.02f', value)
```

## Steps to reproduce the problem

See examples above

## RuboCop version

```
$ [bundle exec] rubocop -V
0.79.0
```

**Comment by inkstak** [comment]
Sorry for the multiple edits, the issue was sent by mistake before I had time to read me again.

**Comment by marcandre** [comment]
If this is still valid, should definitely be fixed.

**Comment by samrjenkins** [comment]
This behaviour is still present. I've taken a bit of a look into it and a solution isn't too challenging.

But is the proposed behaviour something Rubocop should support?

@inkstak proposes that 
```yml
Style/FormatStringToken:
  EnforcedStyle: template
```
should not report offences for the following Ruby
```ruby
format('%<number>0.02f', number: 1)
format('%{number}', number: 1) }
```
The current behaviour will report an offence on the first line (above) because it is not template format.

It strikes me that this is not an issue with the Rubocop rule. If config enforces template style, a developer should remain consistent in all instances. In the scenario where a developer finds a need to apply type, width, precision, or flags formatting to their strings, they ought to opt for the less restrictive style in their configuration:
```yml
Style/FormatStringToken:
  EnforcedStyle: annotated
```
and refactor away from any use of the template style.

@marcandre, what do you think?

**Comment by marcandre** [comment]
My understanding is that this cop helps to choose between two alternatives with equivalent output. The case `'%<number>0.02f'` having no equivalent, it should never raise an offense. I think having `EnforcedStyle: template` should mean prefer template *when possible*. I do not see a "strict" setting as being useful (and what would the auto-correction be?)

**Comment by samrjenkins** [comment]
@marcandre I see your point in enforcing a style only in instances when there are alternatives. Then again, to me, mixing the use of template and annotated within the same project does feel like a bit of a code smell though. If a developer finds themselves making use of `'%<number>0.02f'`, would it not be best practice to consistently use the annotated style throughout the project?

**Comment by marcandre** [comment]
@samrjenkins: Some people will feel like you, and they have a setting for this. Others will feel like me that "%{expr}" is the natural, default, simplest and clearest way and prefer using that whenever possible. In the rare case where that's not sufficient, then we need to be able to use the other form. This has the advantage of highlighting the difference between a straight conversion to string and a more complex one.

Similarly, some people prefer using double quotes everywhere. Others prefer that single quoted strings indicate a string without interpolations and double quotes are reserved for strings requiring interpolation.

**Comment by samrjenkins** [comment]
@marcandre, I too had thought about the quotes analogy. I think it is a good one 👍 Happy to roll with your suggested behaviour.

To confirm, is the suggestion to _never_ report an offence for strings of the form `'%<number>0.02f'`?

**Comment by marcandre** [comment]
> To confirm, is the suggestion to _never_ report an offence for strings of the form `'%<number>0.02f'`?

Yes

**Comment by KessaPassa** [comment]
@marcandre This PR(https://github.com/rubocop/rubocop/pull/10160) looks good for me.
Is there any blocker?
